src/silicone/database_crunchers/linear_interpolation.py

```diff
@@ -0,0 +1,147 @@
+"""
+Module for the database cruncher which makes a linear interpolator between known values
+"""
+
+from pyam import IamDataFrame
+
+from .base import _DatabaseCruncher
+from ..utils import _get_unit_of_variable, _make_wide_db, _make_interpolator
+
+
+class DatabaseCruncherLinearInterpolation(_DatabaseCruncher):
+    """
+    Database cruncher which fits via a linear interpolator to return values from that
+    set of scenarios. Uses mean values in the case of repeated leader values.
+    Returns the follower values at the extreme leader values for leader values more
+    extreme than that found in the input data.
+    """
+
+    def derive_relationship(
+        self, variable_follower, variable_leaders, required_scenario="*"
+    ):
+        """
+        Derive the relationship between two variables from the database.
+
+        Parameters
+        ----------
+        variable_follower : str
+            The variable for which we want to calculate timeseries (e.g.
+            ``"Emissions|CH4"``).
+
+        variable_leaders : list[str]
+            The variable(s) we want to use in order to infer timeseries of
+            ``variable_follower`` (e.g. ``["Emissions|CO2"]``).
+
+        required_scenario : str or list[str]
+            The string which all accepted scenarios are required to match. This may have
+            *s to represent wild cards. It defaults to accept all scenarios.
+
+        Returns
+        -------
+        :obj:`func`
+            Function which takes a :obj:`pyam.IamDataFrame` containing
+            ``variable_leaders`` timeseries and returns timeseries for
+            ``variable_follower`` based on the derived relationship between the two.
+            Please see the source code for the exact definition (and docstring) of the
+            returned function.
+
+        Raises
+        ------
+        ValueError
+            There is no data of the appropriate type in the database.
+             There may be a typo in the SSP option.
+        """
+        if len(variable_leaders) != 1:
+            raise NotImplementedError(
+                "Having more than one `variable_leaders` is not yet implemented"
+            )
+        use_db = self._db.filter(
+            variable=[variable_leaders[0], variable_follower],
+        )
+        if use_db.data.empty:
+            raise ValueError(
```

Comment:
Move to `DatabaseCruncherScenarioAndModelSpecificInterpolate`

Reply:
I've reduced the error message here and put a catch in DatabaseCruncherScenarioAndModelSpecificInterpolate too, it's still worth checking that we have leader/follower data here I think